ocaml/sdk-gen/powershell/autogen/src/Connect-XenServer.cs

```diff
@@ -277,76 +252,49 @@ protected override void ProcessRecord()
                 WriteObject(newSessions.Values, true);
         }
 
-        #region Messages
-
-        const string CERT_HAS_CHANGED_CAPTION = "Security Certificate Changed";
-
-        const string CERT_CHANGED = "The certificate fingerprint of the server you have connected to is:\n{0}\nBut was expected to be:\n{1}\n{2}\nDo you wish to continue?";
-
-        const string CERT_FOUND_CAPTION = "New Security Certificate";
-
-        const string CERT_FOUND = "The certificate fingerprint of the server you have connected to is :\n{0}\n{1}\nDo you wish to continue?";
-
-        const string CERT_TRUSTED = "The certificate on this server is trusted. It is recommended you re-issue this server's certificate.";
-
-        const string CERT_NOT_TRUSTED = "The certificate on this server is not trusted.";
-
-        #endregion
-
         private void AddCertificate(string hostname, string fingerprint)
         {
             var certificates = CommonCmdletFunctions.LoadCertificates();
             certificates[hostname] = fingerprint;
             CommonCmdletFunctions.SaveCertificates(certificates);
         }
 
-        private readonly object certificateValidationLock = new object();
-
-        private bool ValidateServerCertificate(
-            object sender,
-            X509Certificate certificate,
-            X509Chain chain,
-            SslPolicyErrors sslPolicyErrors)
+        private bool ValidateServerCertificate(object sender, X509Certificate certificate, X509Chain chain, SslPolicyErrors sslPolicyErrors)
         {
             if (sslPolicyErrors == SslPolicyErrors.None)
                 return true;
 
-            lock (certificateValidationLock)
+            lock (_certificateValidationLock)
             {
-                bool ignoreChanged = NoWarnCertificates || (bool)GetVariableValue("NoWarnCertificates", false);
-                bool ignoreNew = ignoreChanged || NoWarnNewCertificates || (bool)GetVariableValue("NoWarnNewCertificates", false);
+                bool ignoreChanged = Force || NoWarnCertificates || (bool)GetVariableValue("NoWarnCertificates", false);
+                bool ignoreNew = Force || NoWarnNewCertificates || (bool)GetVariableValue("NoWarnNewCertificates", false);

```

Comment:
I think the `ignoreChanged` here was not correct (in XC at least we treat them separately).